src/backend/gl/src/command.rs

```diff
@@ -636,17 +673,28 @@ impl command::RawCommandBuffer<Backend> for RawCommandBuffer {
         unimplemented!()
     }
 
-    fn copy_buffer_to_image<T>(
-        &mut self,
-        _src: &n::Buffer,
-        _dst: &n::Image,
-        _dst_layout: image::ImageLayout,
-        _regions: T,
-    ) where
-        T: IntoIterator,
-        T::Item: Borrow<command::BufferImageCopy>,
-    {
-        unimplemented!()
+     fn copy_buffer_to_image<T>(
+         &mut self,
+        src: &n::Buffer,
+        dst: &n::Image,
+        _: image::ImageLayout,
+        regions: T,
+     ) where
+         T: IntoIterator,
+         T::Item: Borrow<command::BufferImageCopy>,
+     {
+        let image_raw = match dst {
+            &n::Image::Surface(s) => s,
+            &n::Image::Texture(t) => t
+        };
+
+        let regs = regions.into_iter().collect::<Vec<_>>();
```

Comment:
why do we need to collect here as opposed to just iterating?

Reply:
It's really only used for the assertion that the iterator isn't empty before submitting any commands. We could change it to assert at least one command was submitted instead, or something else?

Reply:
do we have to assert on the empty list?

Reply:
I was going by [`regionCount must be greater than 0`](https://www.khronos.org/registry/vulkan/specs/1.0/man/html/vkCmdCopyBufferToImage.html) but I don’t know if we should be enforcing it somewhere else?